torch/utils/data/dataset.py

```diff
@@ -260,6 +265,116 @@ def __len__(self):
         return len(self.indices)
 
 
+class ChunkDataset(IterableDataset):
+    r"""Dataset which uses hierarchical sampling for efficient data reading.
+
+
+    ``ChunkDataset`` is a stateful dataset that supports hierarchical sampling and
+    efficient reading through chunks. A chunk is selected based on a sampling strategy
+    (first level) and its content is shuffled (second level) before producing
+    a batch or an example.
+
+    In this context, chunks could be files, contents of folder,
+    sections of large text file, etc. For better efficiency, chunks should have roughly
```

Comment:
In distributed training, to eliminate different workers reading uneven amount of data, the chunks need to be roughly the same size. Please consult ``ChunkDataReader`` for more details.

Reply:
fixed